chart/templates/networkpolicy.yaml

```diff
@@ -78,21 +78,25 @@ spec:
     matchLabels:
       release: {{ .Release.Name }}
   egress:
-    # Allows outgoing connections to all pods with
-    # port 443, 8443 or 6443. This is needed for host Kubernetes
-    # access
+    # Allows outgoing connections to all necessary pods with
+    # port 443, 8443 or 6443 or system dns. This is needed for
+    # host Kubernetes access.
     - ports:
         - port: 443
         - port: 8443
         - port: 6443
-    # Allows outgoing connections to all vcluster workloads
-    # or kube system dns server
+        - port: 53
+          protocol: UDP
+        - port: 53
+          protocol: TCP
     - to:
-        - podSelector: {}
+        - podSelector:
+            matchLabels:
+              release: {{ .Release.Name }}
         - namespaceSelector:
             matchLabels:
               kubernetes.io/metadata.name: 'kube-system'
-          podSelector:
+        - podSelector:
```

Comment:
Is this necessary? Since vCluster should only communicate with pods in the namespace `kube-system` and the labels `k8s-app=kube-dns`, this would change the logic to an `OR` and then basically allow communication with all pods in the kube-system namespace.

Reply:
OK. I misunderstand the grammar of `NetworkPolicyEgressRule`. Just revert back